make.ps1

```diff
@@ -0,0 +1,280 @@
+    if (Test-Path '.\.lock') {
+        Remove-Item -Path '.\.lock' -Recurse -Force | Out-Null
+    }
+    New-Item -Path '.\.lock' -ItemType Directory -Force | Out-Null
+    Write-Log 'Creating .\sync\shared\kubejoin.ps1 mock file to keep Vagrantfile happy'
+    New-Item -Path '.\sync\shared\kubejoin.ps1' -ItemType File -Force | Out-Null
+
+    ###### Linux node
+    Write-Log 'vagrant up controlplane'
+    vagrant up controlplane
+    if ($LASTEXITCODE -ne 0) { throw 'Vagrant error' }
+
+    Write-Log 'vagrant status'
+    vagrant status
+
+    ###### Windows node
+    $count = 1
+    while ($true) {
+        $vagrantStatus = $(vagrant status winw1 | Select-String -Pattern 'winw1' | ForEach-Object { $_ -split '\s{2,}' } | Select-Object -Last 1)
+        Write-Log ('vagrant status winw1 - attempt {0} - status: {1}' -f $count, $vagrantStatus)
+        if ($vagrantStatus -match 'running') {
+            break
+        }
+        Write-Log ('vagrant up winw1 - attempt {0}' -f $count)
+        vagrant up winw1
+        $count += 1
+    }
+
+    # Correct SSH key files permissions, otherwise vagrant ssh will keep prompting
+    # for password what defeats the non-interactive purpose of the whole procedure.
+    Set-PrivateKeyPermissions
+
+    $count = 1
+    while ($true) {
+        Write-Log ('kubectl get nodes | grep winw1 - attempt {0}' -f $count)
+        if (vagrant ssh controlplane -c 'kubectl get nodes' | Select-String -SimpleMatch 'winw1') {
+            break
+        }
+        Write-Log ('vagrant provision winw1 - attempt {0}' -f $count)
+        vagrant provision winw1
+        $count += 1
+    }
+
+    Write-Log 'Creating .\.lock\joined indicator for Vagrantfile'
+    New-Item -Path '.\.lock\joined' -ItemType File -Force
+    Write-Log ('vagrant provision winw1 - attempt {0}' -f $count)
+    vagrant provision winw1
+
+    Write-Log 'Creating .\.lock\cni indicator for Vagrantfile'
+    New-Item -Path '.\.lock\cni' -ItemType File -Force
+
+    Write-Log 'Cluster created'
+    vagrant status
+    vagrant ssh controlplane -c 'kubectl get nodes'
+}
+
+function script:Invoke-Status {
+    Write-Log 'Invoking command: status'
+
+    Write-Log 'vagrant status'
+    vagrant status
+    
+    Write-Log 'kubectl get nodes'
+    vagrant ssh controlplane -c 'kubectl get nodes'
+}
+#endregion Command Functions
+
+#region Main Script
+$commands = @{
+    'clean'      = '0. Start fresh destroying any existing Vagrant machines';
+    'download'   = '1. Download Kubernetes binaries for Linux and Windows (set version in variables.yaml or variables.local.yaml)';
+    'build'      = '2. Optionally, build Kubernetes from sources for Linux and Windows';
+    'run'        = '3. Create and run two-node cluster';
+    'status'     = '4. Check state of Vagrant machines and Kubernetes nodes';
+    'smoke-test' = '5. Run smoke tests';
+    'e2e-test'   = '6. Run end-to-end tests';
+    'help'       = 'Print this message';
+}
```

Comment:
`all: 0-fetch-k8s 1-build-binaries 2-vagrant-up 3-smoke-test 4-e2e-test`

It can be a good idea to name the targets and follow the same standard as the Linux ones.

Reply:
Good idea. Done in https://github.com/kubernetes-sigs/sig-windows-dev-tools/pull/254/commits/0b91656aedca6f536b95cb1301c24633e0167ae4